com.zsmartsystems.zigbee.dongle.telegesis/src/test/java/com/zsmartsystems/zigbee/dongle/telegesis/internal/TelegesisFrameHandlerTest.java

```diff
@@ -168,6 +248,33 @@ public void run() {
         assertEquals(1, eventCapture.size());
     }
 
+    /**
+     * Helper Method to send a command to the frameHandler and wait, till the getPacket() method may have read it
+     * 
+     * @param frameHandler the frame handler to send the command to
+     * @param lastSendCommandTime time when the last command was send
+     * @param processed holder for the boolean to check, whether the message is already processed
+     * @param waitTime holder for the time to wait after message has been send
+     * @param time how long the serial port should wait before sending the response
+     */
+    private void sendCommandAndWait(TelegesisFrameHandler frameHandler, long[] lastSendCommandTime, boolean[] processed, long[] waitTime,
+            long time) {
+        TelegesisSetRegisterBitCommand command = new TelegesisSetRegisterBitCommand();
+        command.setBit(0);
+        command.setRegister(0);
+        command.setPassword("password");
+        command.setState(false);
+
+        waitTime[0] = time;
+        lastSendCommandTime[0] = System.currentTimeMillis();
+        processed[0] = false;
+        frameHandler.queueFrame(command);
+        long timeout = System.currentTimeMillis();
+        while (!processed[0] && System.currentTimeMillis() - timeout < 200) {
```

Comment:
I'm not sure I'm keen on this - such a short timeout (200ms) will likely cause problems with CI when running on CI servers running multiple builds (as we have). This will probably cause occasional failures.

Is there a need to have such a short delay?

I'm also wondering if this helper method is really needed or if we can just use ```Mockito.verify``` to wait for ``` the response?